### PR TITLE
EarlGrey 2 Carthage example

### PR DESCRIPTION
**Pull request by bootstraponline**
image

>  Carthage does not currently support static library schemes, nor are there any plans to introduce their support in the future.
>
> https://github.com/Carthage/Carthage

### CODE REVIEW

**Comment by bootstraponline** [comment]
> Because static libraries add additional problems that must be solved: like how to expose the headers for the projects that use the dependency. Static frameworks support this nicely (it's basically a static library that's bundled with the headers), so there's no reason to support static libraries by themselves.
>
> https://github.com/Carthage/Carthage/issues/2685#issuecomment-455258498

Based on feedback from the Carthage team, I recommend that EarlGrey 2 move the static libs to static frameworks. Otherwise it's not possible for EarlGrey 2 to be properly consumed by Carthage.

**Comment by bootstraponline** [comment]
I added a working Carthage example & documentation readme. Using EarlGrey 2 from Carthage source  is possible today.
`Carthage/Checkouts/EarlGrey/EarlGrey.xcodeproj` is used instead of `Carthage/Build/iOS/`

Using the binaries from build will be possible in the future once EG2 is distributed as a framework.

**Comment by bootstraponline** [comment]
I setup the CarthageExample workflow on bitrise to verify master doesn't break carthage support. I'll activate the workflow after this PR is merged.
image

File: Demo/CarthageExample/CarthageExample.xcodeproj/xcuserdata/medwards.xcuserdatad/xcschemes/xcschememanagement.plist [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
```
Comment (tirodkar) [inline Demo/CarthageExample/CarthageExample.xcodeproj/xcuserdata/medwards.xcuserdatad/xcschemes/xcschememanagement.plist]: You should remove these scheme files.

File: Demo/CarthageExample/CarthageExampleUITests/CarthageExampleUITests.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import XCTest
+
+class CarthageExampleUITests: XCTestCase {
+
+    override func setUp() {
+        continueAfterFailure = false
```
Comment (tirodkar) [inline Demo/CarthageExample/CarthageExampleUITests/CarthageExampleUITests.swift]: Remove

File: Demo/CarthageExample/CarthageExampleUITests/EarlGrey.swift [hunk 3]
```diff
@@ -0,0 +1,177 @@
+//
+// Copyright 2018 Google Inc.
```
Comment (tirodkar) [inline Demo/CarthageExample/CarthageExampleUITests/EarlGrey.swift]: Can't you link to the internal EG.swift file? That'll help us test it with Bitrise.

File: Demo/CarthageExample/CarthageExampleUITests/bridge.h [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#ifndef bridge_h
```
Comment (tirodkar) [inline Demo/CarthageExample/CarthageExampleUITests/bridge.h]: Same for this file.

File: docs/setup.md [hunk 5]
```diff
@@ -169,7 +169,7 @@ Add the following imports:
 #import "AppFramework/Action/GREYAction.h"
 #import "AppFramework/Action/GREYActionBlock.h"
 #import "AppFramework/Action/GREYActions.h"
-#import "AppFramework/Matcher/GREYElementMatcherBlock.h"
+#import "CommonLib/Matcher/GREYElementMatcherBlock.h"
```
Comment (tirodkar) [inline docs/setup.md]: Maybe send a separate PR for this?

Comment (bootstraponline) [reply docs/setup.md]: Done: https://github.com/google/EarlGrey/pull/791

File: Demo/CarthageExample/CarthageExample.xcodeproj/xcuserdata/medwards.xcuserdatad/xcschemes/xcschememanagement.plist (hunk 1 above)
Comment (bootstraponline) [reply Demo/CarthageExample/CarthageExample.xcodeproj/xcuserdata/medwards.xcuserdatad/xcschemes/xcschememanagement.plist]: Done

File: Demo/CarthageExample/CarthageExampleUITests/CarthageExampleUITests.swift (hunk 2 above)
Comment (bootstraponline) [reply Demo/CarthageExample/CarthageExampleUITests/CarthageExampleUITests.swift]: Done

File: Demo/CarthageExample/CarthageExampleUITests/EarlGrey.swift (hunk 3 above)
Comment (bootstraponline) [reply Demo/CarthageExample/CarthageExampleUITests/EarlGrey.swift]: Great feedback! Done.

File: Demo/CarthageExample/CarthageExampleUITests/bridge.h (hunk 4 above)
Comment (bootstraponline) [reply Demo/CarthageExample/CarthageExampleUITests/bridge.h]: Done.

File: Demo/CarthageExample/CarthageExample/AppDelegate.swift [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    func applicationDidBecomeActive(_ application: UIApplication) {
+        // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+    }
+
+    func applicationWillTerminate(_ application: UIApplication) {
+        // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
+    }
+
```
Comment (tirodkar) [inline Demo/CarthageExample/CarthageExample/AppDelegate.swift]: rm these whitespaces please.

File: Demo/CarthageExample/CarthageExample/AppDelegate.swift [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    var window: UIWindow?
+
+
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        // Override point for customization after application launch.
+        return true
+    }
+
+    func applicationWillResignActive(_ application: UIApplication) {
+        // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
```
Comment (tirodkar) [inline Demo/CarthageExample/CarthageExample/AppDelegate.swift]: nix the boilerplate comments + methods.

File: Demo/CarthageExample/CarthageExample/Assets.xcassets/AppIcon.appiconset/Contents.json [hunk 8]
```diff
@@ -0,0 +1,98 @@
+{
```
Comment (tirodkar) [inline Demo/CarthageExample/CarthageExample/Assets.xcassets/AppIcon.appiconset/Contents.json]: Can I ask if these can be reused from FunctionalTests?

File: Demo/CarthageExample/CarthageExample/Assets.xcassets/Contents.json [hunk 9]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (tirodkar) [inline Demo/CarthageExample/CarthageExample/Assets.xcassets/Contents.json]: Same.

File: Demo/CarthageExample/CarthageExample/ViewController.swift [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import UIKit
+
+class ViewController: UIViewController {
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view, typically from a nib.
+    }
+
```
Comment (tirodkar) [inline Demo/CarthageExample/CarthageExample/ViewController.swift]: nix any unneeded code.

File: Demo/CarthageExample/readme.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+#import "CommonLib/DistantObject/GREYHostApplicationDistantObject.h"
+#import "CommonLib/Matcher/GREYMatcher.h"
+#import "TestLib/AlertHandling/XCTestCase+GREYSystemAlertHandler.h"
+#import "TestLib/EarlGreyImpl/EarlGrey.h"
+```
+  -  Add `@loader_path/Frameworks` to your `Runpath Search Paths` for both the App and Test Component.
+
+# First test
+
+Write your first test in Swift
```
Comment (tirodkar) [inline Demo/CarthageExample/readme.md]: Swift. *

**Comment by bootstraponline** [comment]
The PR has been open so long, GitHub lost the association and it's no longer updating. 

I submitted https://github.com/google/EarlGrey/pull/798 with all the fixes
